adagrams/game.py

```diff
@@ -1,11 +1,111 @@
+
+import random
+import string
+
 def draw_letters():
-    pass
+    letter_pool = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    hand = []
+    random_letter = ""
+    while len(hand) < 10:
+        random_letter = random.choice(string.ascii_uppercase)
+        if letter_pool[random_letter] > 0:
+            letter_pool[random_letter] = letter_pool[random_letter] - 1
+            hand.append(random_letter)
+        
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    bank = []
+    word = word.upper()
+    for letter in letter_bank:
+        bank.append(letter)
+    for letter in word:
+        if letter in bank:
+            bank.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    chart = {'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+    }
```

Comment:
This is another dictionary we could think about moving making a global function! When we have a larger collection like this, it's possible it may be used in other functions as we expand this program.